### PR TITLE
Support build stages and .mono_repo.yaml

### PR DESCRIPTION
**Pull request by jakemac53**
Fixes https://github.com/dart-lang/mono_repo/issues/50

### CODE REVIEW

File: mono_repo/lib/src/mono_config.dart [hunk 1]
```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
```
Comment (natebosch) [inline mono_repo/lib/src/mono_config.dart]: [nit] date

File: mono_repo/lib/src/mono_config.dart [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    if (unrecognizedKeys.isNotEmpty) {
+      throw new ArgumentError(
+          'Unrecognized keys $unrecognizedKeys in .mono_repo.yaml.'
+          'Only $_validKeys are allowed.');
+    }
+
+    var sdks = (monoYaml['dart'] as List<String>)?.toList();
+    if (sdks == null || sdks.isEmpty) {
+      throw new ArgumentError(
+          'At least one SDK version is required under "dart".');
```
Comment (natebosch) [inline mono_repo/lib/src/mono_config.dart]: would it be reasonable to default to "dev"?

File: mono_repo/lib/src/mono_config.dart (hunk 1 above)
Comment (jakemac53) [reply mono_repo/lib/src/mono_config.dart]: done

File: mono_repo/lib/src/mono_config.dart (hunk 2 above)
Comment (jakemac53) [reply mono_repo/lib/src/mono_config.dart]: We didn't support this previously, I think the main issue is its not clear whether the default should be "dev" or "stable"?

Comment (natebosch) [reply mono_repo/lib/src/mono_config.dart]: Requiring it to be explicit is fine with me.

**Comment by kevmoo** [comment]
```dart ../mono_repo/mono_repo/bin/mono_repo.dart travis
Unhandled exception:
Invalid argument(s): Unrecognized keys [branches, cache, dart_task, language] in .mono_repo.yaml. Only [dart, stages] are allowed.
#0      new MonoConfig.parse (package:mono_repo/src/mono_config.dart:34:7)
#1      getMonoConfigs (package:mono_repo/src/utils.dart:107:15)
```

Should include the relative path so we know which `.mono_repo.yaml` file you're talking about

**Comment by jakemac53** [comment]
> Should include the relative path so we know which .mono_repo.yaml file you're talking about

done

File: mono_repo/.mono_repo.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+dart:
```
[the remark this answers is not in the document]
Comment (jakemac53) [reply mono_repo/.mono_repo.yml]: done

**Pull request merged**